src/lib/circuit_string.ts

```diff
@@ -100,10 +130,10 @@ export class CircuitString extends CircuitValue {
     return new CircuitString(characters);
   }
 
-  private _contains(str: CircuitString): Bool {
+  private _contains(str: CircuitString, size: number): Bool {
     let ret = new Bool(false);
-    const thisLength = Number(this.length().toString());
-    const strLength = Number(str.length().toString());
+    const thisLength = Number(this.usedLength.toString());
+    const strLength = size;
     let i = 0;
 
     while (i + strLength <= thisLength) {
```

Comment:
like in `append`, here you are fixing the circuit to the length of the string. Again I think that could be avoided